packages/react-instantsearch/src/core/createInstantSearchManager.js

```diff
@@ -41,6 +41,11 @@ export default function createInstantSearchManager({
     searching: false,
   });
 
+  function updateClient(client) {
+    helper.setClient(client);
+    search(); // Will it work?
```

Comment:
Note for later: I should remove my own questioning.

Joke aside, I haven't done a proper and complete test on this.

Also I would be interested in your opinion here @vvo @mthuret, should we trigger a new search when the client or the indexName changes. (if not I will cry because I won't know how to test). FYI, we do trigger a new search for new <Configure /> values.